### PR TITLE
MacOS agent AMI creation using packer.

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description

- MacOS agent's AMI creation using packer.
- Increase `idleTerminationMinutes` to 720 (12 hours) to make sure agents are recycled on idle time after every 12 hours.

### Issues Resolved
Part of: https://github.com/opensearch-project/opensearch-build/issues/2216
Related issue: https://github.com/opensearch-project/opensearch-ci/issues/163

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: packer/jenkins-agent-MacOS-x64.json [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{
+  "variables":{
+    "ami_name":"Jenkins-Agent-MacOS-X64-Mac1Metal",
+    "os_version": "12.4",
+    "os_architecture": "x86_64_mac",
+    "build-region":"us-east-1",
+    "build-vpc":"vpc-<>",
+    "build-subnet":"subnet-<>",
+    "build-secgrp":"sg-<>",
```
Comment (gaiksaya) [inline packer/jenkins-agent-MacOS-x64.json:9]: Are we going to replace this or is this how it is supposed to be?

File: packer/README.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 * jenkins-agent-win2016-x64.json: Windows 2016 Server.
 * jenkins-agent-win2019-x64.json: Windows 2019 Server (Recommended).
 * jenkins-agent-win2019-x64-alpine-wsl.json: Windows 2019 Server with WSL enabled running Alpine 3.
+* jenkins-agent-MacOS-x64.json: MacOS with x86_64_mac os_architecture.
```
Comment (peterzhuamazon) [inline packer/README.md:13]: Can you make everything lower case in the file name just to be consistent.

Comment (peterzhuamazon) [reply packer/README.md:13]: Also add the version of the os such as `macos12`.

File: packer/jenkins-agent-MacOS-x64.json [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        ],
+        "most_recent":true
+      },
+      "associate_public_ip_address":false,
+      "ssh_username":  "ec2-user", 
+      "ssh_timeout": "3h",
+      "tenancy": "host",
+      "ebs_optimized": "true",
+      "tags":{
+        "Name": "{{user `ami_name`}}-{{user `build-time`}}",
```
Comment (peterzhuamazon) [inline packer/jenkins-agent-MacOS-x64.json:46]: Please add the OS_VERSION as part of the Name tag.

**Review by peterzhuamazon — APPROVED** [review]
Thanks @prudhvigodithi this is a great setup just a few questions.

File: lib/compute/agent-node-config.ts [hunk 4]
```diff
@@ -210,6 +210,7 @@ export class AgentNodeConfig {
        ebsOptimized: true,
        hostKeyVerificationStrategy: 'OFF',
        iamInstanceProfile: this.AgentNodeInstanceProfileArn,
+       idleTerminationMinutes: "720",
```
[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply lib/compute/agent-node-config.ts:213]: Hey @gaiksaya for Mac the working of an agent is completely differnt, unlike regular ec2 instances the termination and startup is not that quick as it used backend dedicated hosts, to terminate and again start an instance the scrubbing workflow will more take more than an hour, with limited dedicated hosts at the backend, we cant just keep terminating the agents, More info: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-mac-instances.html
Hence recycling once in every 12 hours, to make sure that waiting is reduced.

File: packer/jenkins-agent-MacOS-x64.json (hunk 1 above)
Comment (prudhvigodithi) [reply packer/jenkins-agent-MacOS-x64.json:9]: These values will be replaced per specific account.

File: packer/README.md (hunk 2 above)
Comment (prudhvigodithi) [reply packer/README.md]: Pushed in my latest commit @peterzhuamazon

**Pull request merged**